modules/images/fetchpriority/load.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Module Name: Fetchpriority
+ * Description: Adds `fetchpriority=high` parameter to the non lazy loaded images.
+ * Experimental: Yes
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+/**
+ * Filters an image tag in content to add the fetchpriority attribute if it is not lazy-loaded.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $filtered_image The image tag to filter.
+ * @param string $context        The context of the image.
+ * @return string The filtered image tag.
+ */
+function fetchpriority_img_tag_add_attr( $filtered_image, $context ) {
+
+	if ( 'the_content' !== $context && 'the_post_thumbnail' !== $context ) {
+		return $filtered_image;
+	}
+
+	if ( strpos( $filtered_image, ' loading="lazy"' ) === false && strpos( $filtered_image, ' fetchpriority=' ) === false ) {
+		$filtered_image = str_replace( '<img ', '<img fetchpriority="high" ', $filtered_image );
```

Comment:
If my post contains text and at the end of the article I use an image, then do we also need to add `fetchpriority="high"` attribute? The image is not in view port.

Reply:
I am letting the loading="lazy" code work if to add the as don't what to have both this is an else, and I suspect that when add it to code, it will be mixed into the code that adds loading="lazy"

Reply:
> If my post contains text and at the end of the article I use an image, then do we also need to add fetchpriority="high" attribute? The image is not in view port.

The short answer is no. However, we are expecting you will benefit from marking your first image - based on the expectation that on balance, it will be your LCP element.